packages/ckeditor5-dev-utils/lib/bundler/getplugins.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const path = require( 'path' );
+
+/**
+ * Transforms specified an array of plugin paths to an object contains plugin names
+ * and paths to the plugins.
+ *
+ * Names of plugins returned in the object will be always unique.
+ *
+ * In case of two ore more plugins will have the same name:
+ *   - typing => TypingPlugin
+ *   - path/to/other/plugin/typing => Typing1Plugin
+ *
+ * @param {Array.<String>} pluginPaths
+ * @returns {Object}
+ */
+module.exports = function getPlugins( pluginPaths ) {
+	const plugins = {};
+
+	pluginPaths.forEach( ( pathToFile ) => {
+		const basePluginName = capitalize( path.basename( pathToFile, '.js' ) );
+		let pluginName = basePluginName + 'Plugin';
```

Comment:
> Sorry, but does not sound like a metric for me. ;/

Well, this is the most important metric of a good code. If it's understandable to **other** developers.